Estebanreyl/container registry runtime sdk

estebanreyl opened the pull request:
Should finally address : https://github.com/Azure/azure-rest-api-specs/issues/6457

Readying a new release of the container registry runtime sdk along with brand new tests for all methods

History for Review (avoids showing most auto generated files):
https://github.com/Azure/azure-sdk-for-net/pull/7081/files/f957010c20a205aba4aae265a577799394ba9bae..97ff4d756099e72f3034938b720c4eef21b7f4b8?file-filters%5B%5D=.cs

Included:
- An Example program (Would probably be best to break apart and instead use for documentation purposes?)
- Functional Unit tests for 100% of the paths
- A custom authentication file for our own oauth2

Commit:
Added autogenerated sdk from autorest in order to get a head start with writing tests.

Updated nugget dependencies and added example file (probably will be removed later)

Commit:
Began cleaning up tests

Commit:
Made it possible to run the tests. Some issues remain however. Investigating.

Commit:
Fixed incorrect Unit tests

Commit:
Generated SDK from most recent released swagger.

Commit:
Removed vscode paths from git tracking

Commit:
Began rewriting all unit tests

Commit:
Began adding example v1 Manifest

Commit:
Further work on testing.

Commit:
Begining working on non recorded tests

Commit:
Fixed several more tests 13/20 work

Commit:
Fixes all Manifest and Repository tests.

Commit:
All Tests work. Acquisition of AAD tokens programatically remains troublesome.

Commit:
Fixed AAD authentication issues - Fully functional tests

Commit:
Merge remote-tracking branch 'upstream/master' into estebanreyl/container-registry-sdk

Commit:
Removed odd AssemblyOriginatorKeyFile

Commit:
updated internal files for pipeline tests

estebanreyl commented on sdk/containerregistry/Microsoft.Azure.ContainerRegistry/tests/Helpers/ACRTestUtil.cs:
```diff
@@ -12,37 +12,64 @@ namespace ContainerRegistry.Tests
     using Microsoft.Rest;
     using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
     using Microsoft.Azure.ContainerRegistry;
+    using System.Threading;
```
As in alphabetically?

nathana1 replied:
yes, visual studio has a command for it

estebanreyl commented on sdk/containerregistry/Microsoft.Azure.ContainerRegistry/tests/Tests/ManifestTests.cs:
```diff
@@ -9,77 +9,239 @@ namespace ContainerRegistry.Tests
-            ImageName = ACRTestUtil.ProdRepository,
-            Registry = ACRTestUtil.ManagedTestRegistryFullName,
-            ManifestAttributes = new AcrManifestAttributesBase()
+            Registry = "azuresdkunittest.azurecr.io",
```
For the overall ExpectedAttributesOfProdRepository like objects or its individual fields?  I can do both as well.

nathana1 replied:
most of the registry names, repositories and tags seem like they are defined in the test utils file so if you can just use those values as appropriate

Commit:
Cleanup and regenerated tests

- Includes one more attempted pipeline fix (Returning an aad fake token in the absense of record mode).

Commit:
Updated conditions for pipeline aad acquisition

Commit:
RanNet461 - Generated dlls

Commit:
Further Cleanup

Commit:
Added SessionRecords for net461 folder

Commit:
Removes all files that feel really wrong

Commit:
Remove improper package imports

Commit:
Update Tests.csproj. Pipeline Fix

estebanreyl commented:
@erich-wang , @isra-fel Any chance I could get an update on what else, if anything, may be missing to get this SDK going?

Commit:
Updated version and description

Commit:
Proper jwt token conflict resolution

Commit:
Revert "Proper jwt token conflict resolution"

This reverts commit f525b4cbc161de2f7bfaa4729d391d592ba77b95.

erich-wang commented:
@AlexGhiondea, could you take a look since it is data plane related PR?

Commit:
Removed unecessary updates imports

estebanreyl commented:
I know its been some time and there's other work but is there any chance we can get this reviewed in the nearby future? V2 is getting really close and its been 2 weeks since the PR was ready.

estebanreyl commented:
Will close this in favor of releasing V2 since REST spec just went out,
